coap-gateway/schema/device/status/status.go

```diff
@@ -0,0 +1,30 @@
+package status
+
+import "time"
+
+const ResourceType string = "x.cloud.device.status"
+const Title string = "Device cloud status"
+
+var Interfaces = []string{"oic.if.baseline"}
+var ResourceTypes = []string{ResourceType}
+
+const Href = "/plgd.dev/s"
+
+// Status is resource published by OCF Cloud.
+// - signup: resource published
+// - signin: content changed -> online true and set expires to unix timestap when online state expires (0 means never)
+// - signout/close connection: content changed -> online false
+type Status struct {
+	ResourceTypes []string `json:"rt"`
+	Interfaces    []string `json:"if"`
+	Online        bool     `json:"online"`
+	Expires       int64    `json:"expires"`
+}
+
+// IsOnline evaluate online state
+func (s Status) IsOnline() bool {
+	if !s.Online || s.Expires <= 0 {
+		return s.Online
+	}
+	return time.Now().Before(time.Unix(s.Expires, 0))
```

Comment:
Do you have a test for this? Can it be negative?

Reply:
It cannot be negative .